README.md

```diff
@@ -1 +1,312 @@
+
+## What you’ll build 
+In this guide, you will build a Ballerina based AWS Lambda function which listens to events from a specific S3 bucket in order to detect new object creation. This bucket is used to upload images by various sources, and the job of the lambda is to check if the images have any text, and if so, extract the text from the document, and send an email with the extracted text to the administrator. 
+
+![Ballerina AWS Lambda Deployment](images/ballerina-aws-lambda.svg "Ballerina AWS Lambda Deployment")
+
+## Compatibility
+| Ballerina Language Version | 
+| -------------------------- | 
+| 0.990.3                    |
+
+## Prerequisites
+ 
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install one of the following Ballerina IDE plugins: [VSCode](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina), [IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina)
+- Sign up for an [AWS](https://aws.amazon.com/) account
+- Install and configure [AWS CLI Tools](https://docs.aws.amazon.com/codedeploy/latest/userguide/getting-started-configure-cli.html)
+
+### AWS Access/Secret Key Generation
+
+Now having an active AWS account. You will need to create an access key / secret key pair. The instruction in doing this can be found [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html). Note down the access key and the secret key in a secured place. 
+
+### AWS Lambda Role Generation
+
+- Open the [roles](https://console.aws.amazon.com/iam/home?#/roles) page in the IAM console.
+- Select "Create role".
+- Create roles with the following properties:
+  * Trusted entity - Lambda
+  * Permissions - AWSLambdaBasicExecutionRole
+  * Role name - lambda-role
+
+### Obtaining Tokens for the GMail Connector
+
+- The steps on how to get access to the tokens can be found in the Gmail connector [documentation](https://central.ballerina.io/wso2/gmail).
+
+## Implementation
+
+As the first step, you can build the basic skeleton that is needed for Ballerina to expose as AWS Lambda functions. Let's name the file `image-text-processor.bal`.
+
+```ballerina
+import ballerinax/awslambda;
+
+@awslambda:Function
+public function processImages(awslambda:Context ctx, json input) returns json|error {
+    return "OK";
+}
+```
+
+In Ballerina, any function can be exposed as an AWS Lambda function, by annotating a function which has a similar function signature as the above function with `awslambda:Function`. The `awslambda:Context` object contains contextual information on the current execution, such as the request id, tracing information, and the remaining execution time. The payload is passed as a `json` value in the `input` variable, and the function can return a `json` as a response, or else an `error` can be returned to signal an error situation to the Lambda runtime. 
```

Comment:
```suggestion
In Ballerina, any function can be exposed as an AWS Lambda function by annotating its signature with `awslambda:Function`. The `awslambda:Context` object contains contextual information of the current execution such as the request ID, tracing information, and the remaining execution time. 

The payload is passed as a `json` value via the `input` variable. The function can return a `json` as a response, or else an `error` can be returned to signal of an error situation to the Lambda runtime. 
```

Reply:
You don't say annotating its signature, that is not technically correct. Let me incorporate the other changes suggested.